contracts/modules/GoldenEggMetadata.sol

```diff
@@ -2,38 +2,83 @@
+    //                            STORAGE
+    // =============================================================
+
+    /**
+     * @dev The maximum `tokenId` for `edition` that has a numbered json.
+     * If zero, all `tokenId`s have number jsons.
+     */
+    mapping(address => uint256) public numberedUpto;
+
+    // =============================================================
+    //               PUBLIC / EXTERNAL WRITE FUNCTIONS
+    // =============================================================
+
+    /**
+     * @inheritdoc IGoldenEggMetadata
+     */
+    function setNumberedUpto(address edition, uint256 tokenId) external onlyEditionOwnerOrAdmin(edition) {
+        numberedUpto[edition] = tokenId;
+        emit NumberUptoSet(edition, tokenId);
+    }
+
+    // =============================================================
+    //               PUBLIC / EXTERNAL VIEW FUNCTIONS
+    // =============================================================
+
     /**
      * @inheritdoc IGoldenEggMetadata
      */
     function tokenURI(uint256 tokenId) external view returns (string memory) {
-        uint256 goldenEggTokenId = getGoldenEggTokenId(ISoundEditionV1(msg.sender));
+        uint256 goldenEggTokenId = getGoldenEggTokenId(msg.sender);
         string memory baseURI = ISoundEditionV1(msg.sender).baseURI();
 
-        if (tokenId == goldenEggTokenId) {
-            return bytes(baseURI).length != 0 ? string.concat(baseURI, "goldenEgg") : "";
-        }
+        if (bytes(baseURI).length == 0) return "";
+
+        if (tokenId == goldenEggTokenId) return string.concat(baseURI, "goldenEgg");
 
-        return bytes(baseURI).length != 0 ? string.concat(baseURI, LibString.toString(tokenId)) : "";
+        uint256 n = numberedUpto[msg.sender];
+        return string.concat(baseURI, LibString.toString(n != 0 && tokenId > n ? 0 : tokenId));
     }
 
     /**
      * @inheritdoc IGoldenEggMetadata
      */
-    function getGoldenEggTokenId(ISoundEditionV1 edition) public view returns (uint256 tokenId) {
-        uint256 totalMinted = edition.totalMinted();
-        uint256 mintRandomness = edition.mintRandomness();
+    function getGoldenEggTokenId(address edition) public view returns (uint256 tokenId) {
+        uint256 editionMaxMintable = ISoundEditionV1(edition).editionMaxMintable();
+        uint256 mintRandomness = ISoundEditionV1(edition).mintRandomness();
 
         // If the `mintRandomness` is zero, it means that it has not been revealed,
         // and the `tokenId` should be zero, which is non-existent for our editions,
         // which token IDs start from 1.
         if (mintRandomness != 0) {
-            // Calculate number between 1 and `totalMinted`.
+            // Calculate number between 1 and `editionMaxMintable`.
             // `mintRandomness` is set during `edition.mint()`.
-            tokenId = (mintRandomness % totalMinted) + 1;
+            tokenId = (mintRandomness % editionMaxMintable) + 1;
```

Comment:
won't this pick some random number across the whole uint32.max range?

Reply:
If the mint has not concluded, the golden egg won't be revealed, as the `mintRandomness` is zero. So it doesn't matter here if the value fluctuates, when the mint has not concluded.

After the mint has concluded, the `editionMaxMintable` is fixed.

Reply:
what is the reason to change this from totalMInted to editionMaxMintable?

Reply:
`totalMinted` is the number from ERC721A, which will be allowed to go beyond `editionMaxMintable` in the case of our future bonding curve mints.